packages/teams-js/src/public/appNotification.ts

```diff
@@ -0,0 +1,145 @@
+   * @throws Error if invalid url is passed
+   * @returns void
+   */
+  function validateNotificationDisplayParams(notificationDisplayparam: NotificationDisplayParam): void {
+    const maxTitleLength = 20;
+    const maxContentLength = 40;
+    if (!notificationDisplayparam.title) {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS, message: 'Must supply  notification title to be displayed' };
+    }
+    if (!isValidLength(notificationDisplayparam.title, maxTitleLength)) {
+      throw {
+        errorCode: ErrorCode.INVALID_ARGUMENTS,
+        message: `Invalid notification title length: Maximum title length ${maxTitleLength}, title length supplied ${notificationDisplayparam.title.length}`,
+      };
+    }
+
+    if (!notificationDisplayparam.content) {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS, message: 'Must supply notification content to be displayed' };
+    }
+    if (!isValidLength(notificationDisplayparam.content, maxContentLength)) {
+      throw {
+        errorCode: ErrorCode.INVALID_ARGUMENTS,
+        message: `Maximum content length ${maxContentLength}, content length supplied ${notificationDisplayparam.content.length}`,
+      };
+    }
+
+    if (!notificationDisplayparam.notificationActionUrl) {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS, message: 'Must supply notification url to be displayed' };
+    }
+    if (!isValidUrl(notificationDisplayparam.notificationActionUrl)) {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS, message: 'Invalid url' };
+    }
+
+    if (!notificationDisplayparam.displayDurationInSeconds) {
+      throw {
+        errorCode: ErrorCode.INVALID_ARGUMENTS,
+        message: 'Must supply display duration in seconds to be displayed',
+      };
+    }
+
+    if (notificationDisplayparam.displayDurationInSeconds <= 0) {
+      throw {
+        errorCode: ErrorCode.INVALID_ARGUMENTS,
+        message: 'Notification display time must be greater than zero',
+      };
+    }
+  }
+
+  /**
+   * Displays appNotification after making a validiity check on all of the required parameters, by calling the validateNotificationDisplayParams helper function
+   * An interface object containing all the required parameters to be displayed on the notification would be passed in here
+   * @param notificationdisplayparam - Interface object with all the parameters required to display an appNotificiation
+   * @returns a promise resolution upon conclusion
+   */
+  export function displayInAppNotification(notificationDisplayparam: NotificationDisplayParam): Promise<void> {
+    ensureInitialized(
+      runtime,
+      FrameContexts.content,
+      FrameContexts.stage,
+      FrameContexts.sidePanel,
+      FrameContexts.meetingStage,
+    );
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    validateNotificationDisplayParams(notificationDisplayparam);
+    return sendAndHandleSdkError('appNotification.displayNotification', notificationDisplayparam);
```

Comment:
In our meeting with Trevor, we decided we will be sending a string type of url in notificationDisplayparam to hubSDK.

Reply:
We can talk more about this in our 1:1 today

Reply:
Yes it should be.
I updated this after the meeting, but I made this commit prior to the meeting.